packages/server/src/course/course.controller.ts

```diff
@@ -525,64 +529,87 @@ export class CourseController {
     await this.courseService.removeUserFromCourse(userCourse);
   }
 
-  /* TODO: This is the old login registration form.
-  @Post('submit_course')
-  async submitCourse(@Body() body: SubmitCourseParams): Promise<void> {
-    if (body.password !== process.env.APPLY_PASSWORD) {
-      throw new UnauthorizedException(
-        ERROR_MESSAGES.courseController.invalidApplyURL,
-      );
-    }
-
-    const season = body.semester.split(' ')[0];
-    const year = parseInt(body.semester.split(' ')[1]);
-
-    const semester = await SemesterModel.findOne({
-      where: { season, year },
+  @Post('register_courses')
+  @UseGuards(JwtAuthGuard, CourseRolesGuard)
+  @Roles(Role.PROFESSOR)
+  async registerCourses(
+    @Body() body: RegisterCourseParams[],
+    @UserId() userId: number,
+  ): Promise<void> {
+    // obtains the ProfSectionGroupsModel of the professor
```

Comment:
@IrisLiu-00 do you think we will ever run into the condition where two professors are trying to register a course at the same time due to a miscommunication? They go through the entire flow and reach this step

What if we added a check to ensure the CRN mappings haven't already been created?

Reply:
Was thinking that as well, and tingwei also brought up the situation where if a professor is registering multiple courses but one of them is invalid, then if they try re-registering all of them again it would fail?